flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/InternalTimerService.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.streaming.api.operators;
+
+import org.apache.flink.annotation.Internal;
+
+/**
+ * Interface for working with time and timers.
+ *
+ * <p>This is the internal version of {@link org.apache.flink.streaming.api.TimerService}
+ * that allows to specify a key and a namespace to which timers should be scoped.
+ *
+ * @param <N> Type of the namespace to which timers are scoped.
+ */
+@Internal
+public interface InternalTimerService<N> {
+
+	/** Returns the current processing time. */
+	long currentProcessingTime();
+
+	/** Returns the current event time. */
+	long currentWatermark();
```

Comment:
Jip, will rename everything to `currentWatermark()` because that's also what we already have in other places.